Source/SwiftLintFramework/Rules/Lint/UnusedControlFlowLabelRule.swift

```diff
@@ -86,47 +85,85 @@ public struct UnusedControlFlowLabelRule: SubstitutionCorrectableASTRule, Config
+            )
+        }
+    }
+}
+
+private extension UnusedControlFlowLabelRule {
+    final class Visitor: SyntaxVisitor, ViolationsSyntaxVisitor {
+        private(set) var violationPositions: [AbsolutePosition] = []
+
+        override func visitPost(_ node: LabeledStmtSyntax) {
+            if let position = node.violationPosition {
+                violationPositions.append(position)
+            }
+        }
+    }
+
+    private final class Rewriter: SyntaxRewriter, ViolationsSyntaxRewriter {
+        private(set) var correctionPositions: [AbsolutePosition] = []
+        let locationConverter: SourceLocationConverter
+        let disabledRegions: [SourceRange]
+
+        init(locationConverter: SourceLocationConverter, disabledRegions: [SourceRange]) {
+            self.locationConverter = locationConverter
+            self.disabledRegions = disabledRegions
+        }
+
+        override func visit(_ node: LabeledStmtSyntax) -> StmtSyntax {
+            guard let violationPosition = node.violationPosition else {
+                return super.visit(node)
+            }
+
+            let isInDisabledRegion = disabledRegions.contains { region in
+                region.contains(node.positionAfterSkippingLeadingTrivia, locationConverter: locationConverter)
+            }
+
+            guard !isInDisabledRegion else {
+                return super.visit(node)
+            }
 
-    public func validate(file: SwiftLintFile, kind: StatementKind,
-                         dictionary: SourceKittenDictionary) -> [StyleViolation] {
-        return self.violationRanges(in: file, kind: kind, dictionary: dictionary).map { range in
-            StyleViolation(ruleDescription: Self.description,
-                           severity: configuration.severity,
-                           location: Location(file: file, characterOffset: range.location))
+            let newNode = node.statement.withLeadingTrivia(node.leadingTrivia ?? .zero)
+            correctionPositions.append(violationPosition)
+            return visit(newNode).as(StmtSyntax.self) ?? newNode
         }
     }
+}
 
-    public func substitution(for violationRange: NSRange, in file: SwiftLintFile) -> (NSRange, String)? {
-        var rangeToRemove = violationRange
-        let contentsNSString = file.stringView
-        if let byteRange = contentsNSString.NSRangeToByteRange(start: violationRange.location,
-                                                               length: violationRange.length),
-            let nextToken = file.syntaxMap.tokens.first(where: { $0.offset > byteRange.location }) {
-            let nextTokenLocation = contentsNSString.location(fromByteOffset: nextToken.offset)
-            rangeToRemove.length = nextTokenLocation - violationRange.location
+private extension LabeledStmtSyntax {
+    var violationPosition: AbsolutePosition? {
+        let visitor = BreakAndContinueLabelCollector(viewMode: .sourceAccurate)
```

Comment:
is this a good idea? having a separate visitor simplifies the implementation a lot, but not sure if we'd get huge impacts. 

This shouldn't matter that much in practice though because having a `LabeledStmtSyntax` in Swift is pretty rare to being with.